SmartDeviceLinkService.m

```diff
@@ -0,0 +1,1170 @@
+    
+    [self.manager sendRequest:request withResponseHandler:^(SDLPerformInteraction * _Nullable request, SDLPerformInteractionResponse * _Nullable response, NSError * _Nullable error) {
+        if (!response.success.boolValue) {
+            return;
+        }
+        
+        NSUInteger choiceID = response.choiceID.unsignedIntegerValue;
+        if (choiceID != MWChoiceSetChangeUnitChoiceIdImperial && choiceID != MWChoiceSetChangeUnitChoiceIdMetric) {
+            return;
+        }
+
+        UnitType unit = (choiceID == MWChoiceSetChangeUnitChoiceIdImperial) ? UnitTypeImperial : UnitTypeMetric;
+        [WeatherDataManager sharedManager].unit = unit;
+    }];
+}
+
+- (void)createForecastChoiceSetWithList:(NSArray *)forecasts ofType:(MWInfoType)infoType {
+    BOOL isHourlyForecast = [infoType isEqualToString:MWInfoTypeHourlyForecast];
+    
+    NSDateFormatter *dateFormatShow = [[NSDateFormatter alloc] init];
+    dateFormatShow.timeZone = [NSTimeZone timeZoneForSecondsFromGMT:0];
+    dateFormatShow.locale = self.localization.locale;
+    
+    NSDateFormatter *dateFormatSpeak = [[NSDateFormatter alloc] init];
+    dateFormatSpeak.timeZone = [NSTimeZone timeZoneForSecondsFromGMT:0];
+    dateFormatSpeak.locale = self.localization.locale;
+    
+    if (isHourlyForecast) {
+        dateFormatShow.dateFormat = self.localization[@"forecast.hourly.choice.show"];
+        dateFormatSpeak.dateFormat = self.localization[@"forecast.hourly.choice.speak"];
+    } else {
+        dateFormatShow.dateFormat = self.localization[@"forecast.daily.choice.show"];
+        dateFormatSpeak.dateFormat = self.localization[@"forecast.daily.choice.speak"];
+    }
+    
+    SDLCreateInteractionChoiceSet *createChoiceSetRequest = [[SDLCreateInteractionChoiceSet alloc] init];
+    createChoiceSetRequest.interactionChoiceSetID = @(MWChoiceSetIdList);
+
+    NSMutableSet *filenames = [NSMutableSet set];
+    NSMutableArray *choices = [NSMutableArray array];
+    for (Forecast *forecast in forecasts) {
+        SDLChoice *choice = [[SDLChoice alloc] init];
+        if (self.graphicsAvailable && forecast.conditionIcon != nil) {
+            NSString *filename = forecast.conditionIcon;
+            SDLImage *image = [[SDLImage alloc] initWithName:filename];
+            
+            if ([self.manager.fileManager.remoteFileNames containsObject:filename] == NO) {
+                [filenames addObject:forecast.conditionIcon];
+            }
+
+            choice.image = image;
+        }
+
+        choice.choiceID = @(MWChoiceSetIdList + createChoiceSetRequest.choiceSet.count + 1);
+        choice.menuName = [dateFormatShow stringFromDate:forecast.date];
```

Comment:
Should be `choice.choiceID = @(MWChoiceSetIdList + choices.count + 1);` as `createChoiceSetRequest.choiceSet.count` is always 0